src/store/modules/cosmos/bank/bank.js

```diff
@@ -0,0 +1,232 @@
+			const queryParams = '/' + address + '/' + denom
+			try {
+				const balance = await rootGetters['modules/env/apiClient'].query(
+					queryUrl,
+					queryParams
+				)
+				commit('Balance', { queryParams, balance })
+				if (subscribe) {
+					commit('Subscribe', {
+						action: 'QueryBalance',
+						payload: { address, denom }
+					})
+				}
+			} catch (e) {
+				throw 'API Node unavailable'
+			}
+		},
+		async QueryAllBalances(
+			{ commit, rootGetters },
+			{ address, subscribe = false }
+		) {
+			const queryUrl = '/cosmos/bank/v1beta1/balances'
+			const queryParams = '/' + address
+			try {
+				const balances = await rootGetters['modules/env/apiClient'].query(
+					queryUrl,
+					queryParams
+				)
+				commit('AllBalances', { queryParams, balances })
+				if (subscribe) {
+					commit('Subscribe', {
+						action: 'QueryAllBalances',
+						payload: { address }
+					})
+				}
+			} catch (e) {
+				console.log(e)
+				throw 'API Node unavailable'
+			}
+		},
+		async QueryTotalSupply({ commit, rootGetters }, { subscribe = false }) {
+			const queryUrl = '/cosmos/bank/v1beta1/supply'
+			const queryParams = ''
+			try {
+				const supply = await rootGetters['modules/env/apiClient'].query(
+					queryUrl,
+					queryParams
+				)
+				commit('TotalSupply', { queryParams, supply })
+				if (subscribe) {
+					commit('Subscribe', {
+						action: 'QueryTotalSupply',
+						payload: null
+					})
+				}
+			} catch (e) {
+				throw 'API Node unavailable'
+			}
+		},
+		async QuerySupplyOf({ commit, rootGetters }, { denom, subscribe = false }) {
+			const queryUrl = '/cosmos/bank/v1beta1/supply'
+			const queryParams = '/' + denom
+			try {
+				const amount = await rootGetters['modules/env/apiClient'].query(
+					queryUrl,
+					queryParams
+				)
+				commit('SupplyOf', { queryParams, amount })
+				if (subscribe) {
+					commit('Subscribe', {
+						action: 'QuerySupplyOf',
+						payload: { denom }
+					})
+				}
+			} catch (e) {
+				throw 'API Node unavailable'
+			}
+		},
+		async QueryParams({ commit, rootGetters }, { subscribe = false }) {
+			const queryUrl = '/cosmos/bank/v1beta1/params'
+			const queryParams = ''
+			try {
+				const params = await rootGetters['modules/env/apiClient'].query(
+					queryUrl,
+					queryParams
+				)
+				commit('Params', { params })
+				if (subscribe) {
+					commit('Subscribe', {
+						action: 'QueryParams',
+						payload: {}
+					})
+				}
+			} catch (e) {
+				throw 'API Node unavailable'
+			}
+		},
+		registerTypes() {},
+		/*
+		registerTypes({ dispatch }) {
+			/*
+			Generate type definitions and dispatch action to register them
+
+			const MsgCreatePost = new Type("MsgCreatePost")
+  			.add(new Field("creator", 1, "string"))
+  			.add(new Field("title", 2, "string"))
+				.add(new Field("body", 3, "string"));
+
+			dispatch('modules/wallet/registerType', {typeUrl: '/foo.foo.MsgCreatePost' , type: MsgCreatePost }, { root: true })
```

Comment:
Bank does not require registering message types but left here as a comment to guide auto-generation of other modules.

Because the in-use registry of message types is a property of the signing client instance, this action is dispatched to all modules implementing it whenever the wallet module signifies instantiation of a new signing client. Each module uses that signal to dispatch actions to the wallet to register the custom message types it's using. See: https://github.com/tendermint/vue/pull/49/commits/ed37fa8f68ed7419dc9b4c0dfe1495f3a56e015e